src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -807,6 +808,33 @@ IsReadIntermediateResultFunction(Node *node)
 }
 
 
+/*
+ * FindResultIdOfIntermediateResultFunction extracts the id of the intermediate result
+ * if the given RTE contains a read_intermediate_results function, NULL otherwise
+ */
+Const *
+FindResultIdOfIntermediateResultFunction(RangeTblEntry *rte)
```

Comment:
The name is a little bit confusing. When I read the name, I expect the input to be intermediate result function, but that's not the case.

Maybe: `FindIntermediateResultIdIfExists()` or such could be better

Reply:
done